### PR TITLE
NMS-14844: Documentation for Cloud Services Connector and Time Series DB

### PR DESCRIPTION
**Pull request by Bonrob2**
### All Contributors

* [ ] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14844

### CODE REVIEW

File: docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+. On the Create Your OpenNMS Account screen, type a password in the *Password* box.
+. Click *Create My Account*.
+. Sign in to the OpenNMS Portal with your username and password.
+. Choose *OpenNMS Core Instances* in the left menu and click *Add New*.
+. Type in the name of your instance and click *Generate Activation Key*.
+. Copy the instance key and click *Save & Close*.
+
++
+NOTE: The key is valid for one hour.
+If you do not use it within that time, you must log back into the OpenNMS Portal and copy it from your instance.
```
Comment (bouff) [inline docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc:59]: Users cannot renew/generate another one-time code (such functionality does not exist).

Workaround: they'll have to delete the old instance and create another.

File: docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+. Log in to the OpenNMS {page-component-title} web UI for your instance.
+. In the main menu, click *Info>Cloud Services*.
```
Comment (bouff) [inline docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc:62]: Based on the wording here, it looks like CLOUD-2681 is taken into account.  Can you please confirm @Bonrob2 ?

Comment (Bonrob2) [reply docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc:62]: Yes, the docs take that ticket into account.

File: docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc [hunk 3]
```diff
@@ -0,0 +1,105 @@
+
+= Time Series Service
+
+The time series service provides a scalable cloud-hosted time series database to store and persist performance metrics that {page-component-title} collects from the devices it monitors.
+
+In addition, users can view time series summary data including health status, capacity, and metrics per second in the OpenNMS Portal.
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc:6]: I would suggest combining these paragraphs.

```suggestion
The time series service provides a scalable cloud-hosted time series database to store and persist performance metrics that {page-component-title} collects from the devices it monitors.
In addition, users can view time series summary data including health status, capacity, and metrics per second in the OpenNMS Portal.
```

File: docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+Contact sales@opennms.com to sign up for the time series service.
+
+== Deployment
+
+There are three phases of deployment to set up the time series service for your system:
+
+* Update OpenNMS Cloud Plugin files
+* Obtain an activation key for the service
+* Enable cloud services connectivity
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc:23]:
```suggestion
* Update OpenNMS Cloud Plugin files.
* Obtain an activation key for the service.
* Enable cloud services connectivity.
```

File: docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+=== Update OpenNMS Cloud Plugin files
+
+The OpenNMS Cloud Plugin makes it possible for a {page-component-title} instance to send time series data to an OpenNMS-hosted cloud service.
+
+This plugin is installed by default in {page-component-title} 31.
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc:29]:
```suggestion
This plugin is installed by default in Horizon 31.
```

File: docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+However, you need to move the configuration files into the correct location and restart {page-component-title} as part of the time series service set up:
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc:30]:
```suggestion
However, you need to move the configuration files into the correct location and restart {page-component-title} as part of the time series service setup process:
```

File: docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+[source, console]
+----
+cp /opt/opennms/etc/examples/opennms.properties.d/timeseries.properties /opt/opennms/etc/opennms.properties.d/timeseries.properties
+
+systemctl restart openmns
+----
+
+=== Obtain an activation key for time series services
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc:39]:
```suggestion
=== Obtain an activation key
```

File: docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+You must log in to the OpenNMS Portal to get this activation key, copy it, and then paste it into the appropriate screen in {page-component-title}.
+
+When you signed up for the time series service, you received an activation email from OpenNMS with a login for the OpenNMS Portal.
+
+To obtain the activation key, follow these steps:
+
+. If you have not already done so, activate your OpenNMS Portal account by clicking the *Account Activation* button in your welcome email.
+. On the Create Your OpenNMS Account screen, type a password in the *Password* box.
+. Click *Create My Account*.
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc:51]:
```suggestion
. On the Create Your OpenNMS Account screen, type a password in the *Password* box and click *Create My Account*.
```

File: docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc (hunk 1 above)
Comment (emarsh1995) [inline docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc:59]:
```suggestion
. Copy the instance key and click *Save & Close*.
+
NOTE: The key is valid for one hour.
If you do not use it within that time, you must log back into the OpenNMS Portal and copy it from your instance.
```

File: docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc (hunk 2 above)
Comment (emarsh1995) [inline docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc:62]:
```suggestion
. Click menu:Info[Cloud Services] in the top menu bar.
```

File: docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+----
+
+Check Karaf logs if the registration fails or the health check does not say `SERVING`.
+If this is the case, you will need to open a support ticket.
+
+You can also check that `OPENNMS_HOME/share/rrd` files are still being updated.
+If they are not, this indicates that the time series service is not working.
+
+[[view-time-series-data]]
+== View time series summary data
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc:95]:
```suggestion
== View time series summary data

```

**Review by emarsh1995 — CHANGES_REQUESTED** [review]
Some minor changes. Looks good overall! :)

**Review by drewym — APPROVED** [review]

**Pull request merged**